### PR TITLE
Add JavaInfo.transitive_full_compile_time_jars.

### LINKED ISSUE #4297: JavaInfo should expose `transitive_full_compile_jars`
For some JVM languages (kotlin), we need access to the transitive set of original "full" (non-ABI) jars.  For example consider this code snippet that attempts to build a classpath:

```python
    jars = depset()
    # Populate from (transitive, pretty please) java dependencies.  
    for dep in ctx.attr.java_deps:
        if java_common.provider in dep:
            info = dep[java_common.provider]
            # Here's what we have currently
            jars += info.full_compile_jars
            # Here's what we'd like to do
            jars += info.transitive_full_compile_jars
```

cc @iirina Thanks in advance for considering this!

https://github.com/pubref/rules_kotlin/issues/47

https://github.com/pubref/rules_kotlin/pull/41

### PR DESCRIPTION
**Pull request by uri-canva**
Inspired by https://github.com/bazelbuild/bazel/commit/655a529db693e0d3c89a68ca81fc4858147b90f0.

Fixes #4297.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/rules/java/JavaInfo.java [hunk 1]
```diff
@@ -463,14 +463,29 @@ public SkylarkNestedSet getTransitiveRuntimeJars() {
...
+      doc = "Depset of compile time jars recursively required by this target. See `compile_jars` "
           + "for more details.",
       structField = true
   )
   public SkylarkNestedSet getTransitiveCompileTimeJars() {
     return SkylarkNestedSet.of(Artifact.class, getTransitiveDeps());
   }
 
+  @SkylarkCallable(
+      name = "transitive_full_compile_time_jars",
```
Comment (uri-canva) [inline src/main/java/com/google/devtools/build/lib/rules/java/JavaInfo.java]: The name is a bit of a mouthful, but it mirrors the existing `transitive_compile_time_jars`. I noticed it's also exposed as `transitive_deps`, maybe this should just be `transitive_full_deps`?

File: src/main/java/com/google/devtools/build/lib/rules/java/JavaInfo.java [hunk 2]
```diff
@@ -463,14 +463,29 @@ public SkylarkNestedSet getTransitiveRuntimeJars() {
 
   @SkylarkCallable(
       name = "transitive_compile_time_jars",
-      doc = "Depset of compile time jars recusrively required by this target. See `compile_jars` "
+      doc = "Depset of compile time jars recursively required by this target. See `compile_jars` "
```
Comment (uri-canva) [inline src/main/java/com/google/devtools/build/lib/rules/java/JavaInfo.java]: I can revert this if it's better to not have unrelated changes.

**Comment by uri-canva** [comment]
cc @iirina, @pcj

**Comment by uri-canva** [comment]
giphy

**Comment by jin** [comment]
@iirina, PTAL or reassign? Thanks!

**Comment by cushon** [comment]
@uri-canva did you see my question in the bug?

**Comment by uri-canva** [comment]
No I didn't, I wasn't subscribed to that.

Sorry it turns out I didn't need this as my problem was due to a misunderstanding of how to use the datanucleus bytecode enhancer, the error messages were misleading and this changed fixed the problem I had, but it wasn't the correct way of fixing it.

**Pull request closed without merging**